Sources/Shell/Shell.swift

```diff
@@ -0,0 +1,96 @@
+import Foundation
+
+/// Utility for executing shell commands.
+public enum Shell {
+    /// Error thrown when a process execution fails.
+    public enum Error: Swift.Error, LocalizedError {
+        case nonZeroExit(command: [String], exitCode: Int32)
+        
+        public var errorDescription: String? {
+            switch self {
+            case .nonZeroExit(let command, let exitCode):
+                return "Command '\(command.joined(separator: " "))' failed with exit code \(exitCode)"
+            }
+        }
+    }
+    
+    /// Run a CLI command.
+    ///
+    /// This method executes a command in a subprocess. If the command is not successful
+    /// (indicated by a non-zero exit code), an error is thrown.
+    ///
+    /// - Parameter arguments: An array of command-line arguments to execute.
+    /// - Returns: A string containing the command's standard output and standard error.
+    /// - Throws: An error if the command execution fails
+    @discardableResult public static func run(_ arguments: [String]) async throws -> String {
+        // TODO: It's probably a good idea to separate stdout and stderr streams.
+
+        // Log the command before execution
+        print("Executing: \(arguments.joined(separator: " "))")
```

Comment:
is there some good structured logging library we can use?

Reply:
Yes, we should probably: we should probably just . I imagine we'd:

- Use `print` for normal stdout
- Use [SwiftLog](https://github.com/apple/swift-log) for structured logging, like here